application/models/services/SurveyPermissions.php

```diff
@@ -59,7 +66,9 @@ public function getUserPermissionCriteria()
         $criteria->condition = 't.entity_id =:entity_id';
         $criteria->addCondition('t.entity =:entity');
         $criteria->addNotInCondition('u.uid', [$currentUserId]);
-        $criteria->addInCondition('t.uid', $userList);
+        if (!$hasUserReadPermission && $this->userControlSameGroupPolicy) {
+            $criteria->addInCondition('t.uid', $userList);
```

Comment:
Psalm warning on this line?

Reply:
Came from https://github.com/LimeSurvey/LimeSurvey/blob/de071e01d299b41cc79cc4067f06760ef351da31/application/helpers/common_helper.php#L666 

No user entered value, and return only id.

What is the alert ?